lib/checkunusedvar.cpp

```diff
@@ -1609,17 +1609,31 @@ bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func, const To
     }
 
     bool sideEffectReturnFound = false;
+    std::list<const Variable*> pointersToGlobals;
     for (Token* bodyToken = func.functionScope->bodyStart->next(); bodyToken != func.functionScope->bodyEnd;
-         bodyToken = bodyToken->next()) {
+            bodyToken = bodyToken->next())
+    {
+        // check variable inside function body
         const Variable* bodyVariable = bodyToken->variable();
         if (bodyVariable) {
-            // check variable for side-effects
             if (!isVariableWithoutSideEffects(*bodyVariable)) {
                 return false;
             }
             // check if global variable is changed
-            if (bodyVariable->isGlobal()) {
-                return false; // TODO: analyze global variable usage
+            if (bodyVariable->isGlobal() ||
+                (std::find(pointersToGlobals.begin(), pointersToGlobals.end(), bodyVariable) != pointersToGlobals.end()))
+            {
+                if (isVariableChanged(bodyToken, 20, nullptr, true)) {
+                    return false;
+                }
+                // check if pointer to global variable assigned to another variable (another_var = &global_var)
+                if (Token::simpleMatch(bodyToken->tokAt(-1), "&") && Token::simpleMatch(bodyToken->tokAt(-2), "=")) {
+                    const Token* assigned_var_token = bodyToken->tokAt(-3);
+                    if (assigned_var_token && assigned_var_token->variable())
```

Comment:
if this condition is true then you don't care that the address is used somehow.. would probably be safer to return false then.

Reply:
As I see, `isVariableChanged()` works fine enough with "pointer-to-global" variables. I prefer to leave it as is, if you not insist.
Do you see any test cases which will force with code work not as expected?

Reply:
I think it works well.. right now I am only worried about the todo test case. Do you know it it's complex to fix that?

Reply:
Probably, it should be not so complex. But it can affect existing tests and I will need some time to figure out with them.
Also I can not estimate a free time amount I will be able to put on this work in the nearest time. =)
How about this one - I will return false for now and update astutils.cpp and this piece of code in next PR?

Reply:
It's fine to make a partial fix.. and finish later. but then I feel more comfortable if there are no false positives in the meantime. So well.. could you return false if there is a global array/pointer variable.